frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/IncrementalBuildExecutionDigest.java

```diff
@@ -0,0 +1,214 @@
+package com.github.eirslett.maven.plugins.frontend.lib;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.KeyDeserializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+import static java.util.Objects.requireNonNull;
+
+public class IncrementalBuildExecutionDigest {
+
+    /**
+     * This should be incremented as soon as the digest schema or semantics change
+     */
+    public static final Long CURRENT_DIGEST_VERSION = 1L;
+
+    private static final String SERIALIZATION_CHARSET = UTF_8.toString();
+    /**
+     * Must be something that would be encoded otherwise we could end up with a set of
+     * parts we don't expect.
+     */
+    static final String SERIALIZATION_SEPARATOR = ";";
+
+    public Long digestVersion;
+
+    @JsonDeserialize(keyUsing = ExecutionCoordinatesDeserializer.class)
+    @JsonSerialize(keyUsing = ExecutionCoordinatesSerializer.class)
+    public Map<ExecutionCoordinates, Execution> executions;
+
+    public IncrementalBuildExecutionDigest() {
+            // for Jackson
+    }
+
+    public IncrementalBuildExecutionDigest(Long digestVersion, Map<ExecutionCoordinates, Execution> executions) {
+        this.digestVersion = digestVersion;
+        this.executions = executions;
+    }
+
+    public static class ExecutionCoordinates {
+        public String goal;
+        public String id;
+        public String lifecyclePhase;
+
+        public ExecutionCoordinates() {
+            // for Jackson
+        }
+
+        public ExecutionCoordinates(String goal, String id, String lifecyclePhase) {
```

Comment:
Not sure "Coordinates" is a term relevant to this domain? maybe just "Details"?

Reply:
These should be unique in a project. This allows us to invalidate parts of the file with time.